### ISSUE TITLE
Router animations not working when an md-input is present in the :enter component

### ISSUE THREAD

**Issue by Ploppy3** [angular/angular#17631]
## I'm submitting a ...

<pre><code>
[ ] Regression (behavior that used to work and stopped working in a new release)
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request
</code></pre>

## Current behavior

If a component has an md-input and you try to navigate to it, the animation will not work at all.

## Expected behavior

Router navigation should work properly when navigating to a component with an md-input.

## Minimal reproduction of the problem with instructions
Create an empty component & a component with an md-input.
Create a simple router animation:
```
transition('* => *', [
    query(':leave', [
      animate('.5s ease-in-out', style({opacity: 0}))
    ], {optional: true})
])
```
Navigate between the components.
The animation when navigating from _component => component with an md-input_ will not work.

## Please tell us about your environment
Angular version: 4.2.3

Browser:
- [x] Chrome (desktop) version 59.0.3071.104
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: 6.11.0
- Platform: Windows

**Comment by matsko** [comment]
I have a feeling that #17737 will fix the issue. It's hard to say because there is no demo. I'll report back here once it's in and then you can test against the angular-animations-builds and angular-platform-browser-builds NPM targets.

**Comment by Ploppy3** [comment]
I'll come to you once this patch is released.

**Comment by matsko** [comment]
@Ploppy3 please test using 4.2.5 or 4.3.0-beta.1.

**Comment by Ploppy3** [comment]
Thank you, it fixed it.

**Comment by Ploppy3** [comment]
@matsko Nevermind... It fixes an issue I had with animations. Though animations are still broken as soon as there is an md-input into the :enter component.

**Comment by Ploppy3** [comment]
@matsko 4.2.6 didn't help.

**Comment by Ploppy3** [comment]
@matsko 4.3.0 didn't help.

**Comment by matsko** [comment]
I believe #18210 will fix it

**Comment by matsko** [comment]
Fixed by 5db6f38b73bf4ad55a1e176c718a49c78e530d17

**Comment by Ploppy3** [comment]
@matsko Thanks!
